api/controllers/cart.controller.js

```diff
@@ -1,90 +1,158 @@
+
+        if (!Array.isArray(cart) || cart.length === 0) {
+            return res.status(400).json({ message: 'Cart must be a non-empty array' });
+        }
+
+        //  plant details
+        const plantIds = cart.map(item => item.plantId);
+        const plants = await PlantInfo.find({
+            _id: { $in: plantIds },
+            status: 'PUBLISH',
+            isActive: true,
+        });
+
+        if (plants.length !== cart.length) {
+            return res.status(404).json({ message: 'Some plants are invalid or inactive' });
+        }
+
+        // let customerName = '';
+      
+        // if (customerId) {
+        //     const customer = await Customer.findById(customerId);
+        //     if (!customer) {
+        //         return res.status(404).json({ message: 'Customer not found' });
+        //     }
+        //     // customerName = customer.names.customer.name;
+        // }
+
+        // cart items
+        const cartItems = cart.map(item => {
+            const plant = plants.find(p => p._id.toString() === item.plantId.toString());
+            if (!plant) {
+                throw new Error(`Plant with ID ${item.plantId} not found`);
+            }
+            return {
+                plantId: plant._id,
+                name: plant.names,
+                price: plant.sellingPrice,
+                discountedPrice: plant.discountedSellingPrice || plant.sellingPrice,
+                qty: item.qty,
+                tips: plant.tips.join(', '),
+                moreInfo: plant.moreInfo,
+                tags: plant.tags.map(tag => tag.name),
+            };
+        });
+
+        // Calci
+        let totalAmount = cartItems.reduce((total, item) => total + item.discountedPrice * item.qty, 0);
+
+        let cartData;
+        if (!uuid) {
+            // Create a new cart 
+            cartData = new Cart({
+                items: cartItems,
+                totalAmount,
+                uuid: crypto.randomUUID(), // Generate UUID
+            });
+        } else {
+            // If UUID exists, update the cart
+            cartData = await Cart.findOneAndUpdate(
+                { uuid },
+                { items: cartItems, totalAmount },
+                { new: false } 
+            );
+
+            if (!cartData) {
+                return res.status(404).json({ message: 'Cart not found' });
+            }
+        }
+
+        //  offer if offerId is there
+        if (offerId) {
+            const offer = await Offer.findById(offerId);
+
+            if (!offer) {
+                return res.status(404).json({ message: 'Offer not found' });
+            }
+
+            // Check if offer is active or  not
+            if (!offer.isActive) {
+                return res.status(400).json({ message: 'Offer is not active' });
+            }
+
+            // Check if the offer is applicable to set of the array client
+            const isClient = offer.clients.includes(customerId); // clientId  frome : --------
```

Comment:
this is not isClient
you need to check plantIds 
![image](https://github.com/user-attachments/assets/1aaf33a7-c0ea-45b8-bb73-8b9f8af4169d)